File: test/show-args/check.py
from subprocess import *

compile_cmd = "g++ -std=c++11 show-args.cpp -o show-args"
retcode = call(compile_cmd, shell=True)

run_cmd = "./show-args a b 'c d'> a.txt"
retcode = call(run_cmd, shell=True)

interpret_cmd = "cpps show-args.cpp a b 'c d' > b.txt"
retcode = call(interpret_cmd, shell=True)

compare_cmd = "diff a.txt b.txt"
retcode = call(compare_cmd, shell=True)

if (retcode != 0):
    print "error found!"

File: test/show-args/obsolete/check.cpp
// this is the c++ version of check.py
// you can run it by "cpps check.cpp"
#include <iostream>
#include <cstdlib>

using namespace std;

int main()
{
    int retcode;

    string compile_cmd = "g++ -std=c++11 show-args.cpp -o show-args";
    system(compile_cmd.c_str());

    string run_cmd = "./show-args a b 'c d'> a.txt";
    system(run_cmd.c_str());

    string interpret_cmd = "cpps show-args.cpp a b 'c d' > b.txt";
    retcode = system(interpret_cmd.c_str());

    string compare_cmd = "diff a.txt b.txt";
    retcode = system(compare_cmd.c_str());

    if (retcode != 0)
        cout << "error found!" << endl;


}
